### PR TITLE
Make several refactoring improvements to some `format_*()` functions

### PR DESCRIPTION
**Pull request by rich-iannone**
These changes provide a few more improvements to some `format_*()` functions and related utility functions.

### CODE REVIEW

File: R/utils_formatters.R [hunk 1]
```diff
@@ -307,14 +295,18 @@ context_percent_mark <- function(context) {
...
+                                 currency) {
 
-  switch(context,
-         latex = "\\\\$",
-         "\\$")
-}
+  # If we supply `NULL` as currency, then
+  # return an empty string
+  if (is.null(currency)) {
+    return("")
```
Comment (rich-iannone) [inline R/utils_formatters.R]: Return `NULL` instead.

File: R/utils_formatters.R [hunk 2]
```diff
@@ -361,9 +358,174 @@ paste_currency_str <- function(x,
...
 }
+
+#' @noRd
+format_in_accounting_style <- function(x,
+                                       x_vals,
+                                       accounting,
+                                       minus_mark,
+                                       parens_marks) {
+
+  # Return values unchanged if there are no negative values
```
Comment (rich-iannone) [inline R/utils_formatters.R]: Store `x_vals < 0` as a variable for repeated use.

File: R/utils_formatters.R [hunk 3]
```diff
@@ -361,9 +358,174 @@ paste_currency_str <- function(x,
...
+    )
+
+  # Handle replacement of the minus mark in number
+  # and exponent parts
+  x %>%
+    tidy_gsub("-", minus_mark, fixed = TRUE)
+}
+
+#' @noRd
+num_formatter_factory <- function(context,
```
Comment (rich-iannone) [inline R/utils_formatters.R]: Remove `NULL` defaults.

File: R/utils_formatters.R [hunk 4]
```diff
@@ -361,9 +358,174 @@ paste_currency_str <- function(x,
...
+                                  dec_mark = NULL,
+                                  currency = NULL,
+                                  drop_trailing_zeros = NULL,
+                                  accounting = NULL,
+                                  incl_space = NULL,
+                                  placement = NULL,
+                                  pattern = NULL,
+                                  format_fn = NULL) {
+
+  force(context)
```
Comment (rich-iannone) [inline R/utils_formatters.R]: Explore possibility of `force()`ing vars in a list.

Comment (rich-iannone) [reply R/utils_formatters.R]: `force(list(a, b, c, ...))`

File: R/utils_formatters.R [hunk 5]
```diff
@@ -361,9 +358,174 @@ paste_currency_str <- function(x,
...
+    # for scientific foramtting since their order would be zero
+    small_pos <- has_order_zero(x_vals)
+
+    # Format all non-NA x values with a formatting function
+    x_str_vals <-
+      format_fn(x_vals, decimals, sep_mark, dec_mark, drop_trailing_zeros)
+
+    # Format scientific notation values in a way that
+    # is prettier than the form offered by `formatC()`
+    x_str_vals <-
```
Comment (rich-iannone) [inline R/utils_formatters.R]: Intersperse comments and function calls in piped sequence.

File: R/utils_formatters.R (hunk 4 above)
Comment (rich-iannone) [reply R/utils_formatters.R]: @schloerke this seemed to work in my testing (no R error about syntax). However, do you know of any way to verify that the variables are indeed forced?

Comment (rich-iannone) [reply R/utils_formatters.R]: I tested this to ensure that the forcing works with a list.

**Comment by jcheng5** [comment]
Addendum to 5. in my comment above: rather than forcing all of the number-formatting-related arguments, just force the ones that aren't passed directly to `num_formatter_factory` or whatever; i.e. the ones you use yourself in `format_fn`. It'll make sense after you do step 2.

Is there a reason you didn't push the calls to `get_locale_sep_mark`, `get_locale_dec_mark`, and `normalize_suffixing_inputs` into `num_formatter_factory` as well? I'm not necessarily saying it's wrong the way it is, just curious if there was a specific reason.

**Comment by rich-iannone** [comment]
Thanks for the quick turnaround with the review here! I'll go through these changes but I'll answer one or two of the questions above first.

The `format_num_to_str_e()` and `format_num_to_str_c()` are separated just so we can have targeted tests (which don't yet exist). 

There is no reason that the calls to `get_locale_sep_mark()`, `get_locale_dec_mark()`, and `normalize_suffixing_inputs()` aren't in the factory. I'll work toward putting those in.

**Pull request merged**